src/cpu.c

```diff
@@ -814,8 +814,7 @@ static int cpu_read (void)
 
 	if (pnumcpu != numcpu || perfcpu == NULL)
 	{
-		if (perfcpu != NULL)
-			free(perfcpu);
+		free(perfcpu);
```

Comment:
Hey, please review this!

Code is strange:

if (perfcpu == NULL) 
  free(perfcpu);